newlib/testsuite/newlib.string/tstring.c

```diff
@@ -56,7 +56,18 @@ void myset (char *target, char ch, int size)
 
 int main()
 {
+  /**
```

Comment:
Would making all of these arrays 'static' work around the initialization issue? That would also making this test run in environments without a huge amount of stack space.